shanoir-ng-datasets/src/main/java/org/shanoir/ng/download/DicomJsonUtils.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.download;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * This class is used to download files on using WADO URLs:
+ * 
+ * WADO-RS URLs are supported: http://dicom.nema.org/DICOM/2013/output/chtml/part18/sect_6.5.html
+ * WADO-URI URLs are supported: http://dicom.nema.org/DICOM/2013/output/chtml/part18/sect_6.2.html
+ * 
+ * WADO-RS: http://dcm4chee-arc:8081/dcm4chee-arc/aets/DCM4CHEE/rs/studies/1.4.9.12.22.1.8447.5189520782175635475761938816300281982444
+ * /series/1.4.9.12.22.1.3337.609981376830290333333439326036686033499
+ * /instances/1.4.9.12.22.1.3327.13131999371192661094333587030092502791578
+ * 
+ * As the responses are encoded as multipart/related messages,
+ * this class extracts as well the files contained in the response to
+ * the file system.
+ * 
+ * WADO-URI: http://dcm4chee-arc:8081/dcm4chee-arc/aets/DCM4CHEE/wado?requestType=WADO
+ * &studyUID=1.4.9.12.22.1.8444.518952078217568647576155668816300281982444
+ * &seriesUID=1.4.9.12.22.1.8444.60998137683029030014444439326036686033499
+ * &objectUID=1.4.9.12.22.1.8444.1313199937119266109555587030092502791578
+ * &contentType=application/dicom
+ * 
+ * WADO-URI Web Service Endpoint URL in dcm4chee arc light 5:
+ * http[s]://<host>:<port>/dcm4chee-arc/aets/{AETitle}/wado
+ *
+ * This Spring service component uses the scope singleton, that is there by default,
+ * as one instance should be reused for all other instances, that require usage.
+ * No need to create multiple.
+ * 
+ * @author mkain
+ *
+ */
+public class DicomJsonUtils {
+    
+    public static final String STUDY_INSTANCE_UID = "0020000D";
+    public static final String SERIE_INSTANCE_UID = "0020000E";
+    public static final String OBJECT_INSTANCE_UID = "00080018";
+    public static final String VALUE = "Value";
+    
+    public static final String STUDIES = "studies";
+    public static final String SERIES = "series";
+    public static final String INSTANCES = "instances";
+    
+    public static String inflfateDCM4CheeJSON(String json) throws JSONException {
+        JSONArray flat = new JSONArray(json);
+        
+        // 1st part : create the studies / seies / instances tree
```

Comment:
@michaelkain I agree, this is an experiment I developed during christmas hollidays. I wanted to fetch a whole examination and transform it into a tree like study -> series -> instances because I wanted to be able to know which dicom field is bound to which level and then set finer rules in quality and studycards. I think it works but for the moment it ends up useless. I should remove it.